server/lists/adapters/case-view-all-stages.js

```diff
@@ -38,103 +34,79 @@ const getRequestStatus = (dueDate, { status, decision }) => {
+    }
+    if (contributionBusinessUnit) {
+        return getChoicesValue({ contributionBusinessUnit }, choices);
+    }
 
-async function parseMultipleContributions (value, choices, fromStaticList) {
-    const contributions = JSON.parse(value);
-    const contributionStrings = await getContributionStrings(contributions, choices, fromStaticList);
-    return contributionStrings.join(', ');
-}
+    return getChoicesValue({ contributionBusinessArea }, choices);
+};
+
+const getChoicesValue = (choice, choices) => {
+    const [name, value = ''] = getObjectNameValue(choice);
+
+    const choiceList = choices[name] ?? [];
+    return choiceList.find(item => item.value === value)?.label ?? value;
+};
 
-const renderSomuListItems = async ( { caseType, type, choices }, value, fromStaticList) => {
+const renderSomuListItems = async ( { caseType, type }, choices, { fromStaticList, fetchList, caseId }) => {
     const somuType = await fromStaticList('SOMU_TYPES', [caseType, type]);
 
     if (somuType && somuType.schema && somuType.schema.renderers) {
         const { unallocated } = somuType.schema.renderers;
-
         if (unallocated) {
+            const somuItems = await fetchList('CASE_SOMU_ITEM', {
+                caseId,
+                somuTypeId: somuType.uuid
+            });
+
+            if (somuItems.length === 0) {
+                return;
+            }
+
             switch (unallocated) {
                 case 'MultipleContributions': {
-                    const values = await parseMultipleContributions(value, choices, fromStaticList);
-                    return values;
+                    return parseMultipleContributions(somuItems, choices);
                 }
                 case 'ApprovalRequests': {
-                    const values = await parseApprovalRequests(value, choices, fromStaticList);
-                    return values;
+                    return parseApprovalRequests(somuItems, choices);
                 }
             }
         }
     }
-
-    return value;
 };
 
-module.exports = async (template, { fromStaticList }) => {
+module.exports = async (template, request) => {
     const builder = Form()
         .withTitle(template.caseReference)
         .withNoPrimaryAction();
 
-    const sections =
-        (await Promise.all(Object.entries(template.schema.fields).map(async ([stageId, fields]) => {
-
-            const stageFields = fields.map(
-                fieldTemplate => getComponentFromField(fieldTemplate, template))
-                .filter(component => component !== undefined); // remove empty elements caused by hidden fields .etc
-            const stageName = await fromStaticList('S_STAGETYPES', stageId);
-
-            return { title: stageName, items: stageFields };
-        }))).filter(stage => {
-            return stage.items && stage.items.length > 0;
-        }); // filter out empty sections
-
     const data = (await Promise.all(Object.entries(template.schema.fields)
```

Comment:
This loops around all of the fields regardless of if they are shown or not - resulting in potentially a high cost value for this method. If the same field is present multiple times, this would also cause the same impact. This should be looked at as part of the unallocated accordion rewrite. HOCS-5569